packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -420,12 +420,9 @@ class Analytics implements IAnalytics {
       this.userSessionManager.getUserId() ||
       this.userSessionManager.getAnonymousId();
 
-    // Set the new user ID only after determining the previous ID
-    this.userSessionManager.setUserId(tryStringify(payload.to));
-
     this.eventManager.addEvent({
       type,
-      to: payload.to,
+      to: tryStringify(payload.to),
```

Comment:
this should happen in the eventMethod Overloads file aliasArgumentsToCallOptions method

Reply:
Removed this as @MoumitaM is implementing the changes in the API overloads in another PR.